internal/fileutils/fileutils_test.go

```diff
@@ -246,3 +247,106 @@ func TestCopyFile(t *testing.T) {
 		})
 	}
 }
+
+func TestLrename(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		sourceDoesNotExist     bool
+		destIsFile             bool
+		destIsSymlink          bool
+		destIsDanglingSymlink  bool
+		destIsDir              bool
+		destIsUnreadable       bool
+		destParentDoesNotExist bool
+
+		wantError     bool
+		wantErrorType reflect.Type
```

Comment:
I wouldn't bother reflect here, but rather, use:

`wantErrorEquals error` and then just use `require.ErrorIs`

Reply:
that doesn't work when verifying if an error is of a specific error type instead of a specific error instance.

Reply:
Well it does now that I've re-impemented Is, and it's a safer way IMHO, so we use the same logic everywhere.